client/reader/stream/index.jsx

```diff
@@ -412,6 +413,9 @@ export default class ReaderStream extends React.Component {
 	showUpdates = () => {
 		this.props.onUpdatesShown();
 		FeedStreamStoreActions.showUpdates( this.props.store.id );
+		if ( this.props.recommendationsStore ) {
```

Comment:
It isn't immediately obvious to me why we want to shuffle the posts on `showUpdates()`.
showUpdates is called when we `goToTop` as the result of keyboard navigation (and i'm not sure when else)